src/snowflake/snowpark/session.py

```diff
@@ -207,6 +209,23 @@ def _get_active_sessions() -> Set["Session"]:
 def _add_session(session: "Session") -> None:
     with _session_management_lock:
         _active_sessions.add(session)
+        global _register_atexit
+        if not _register_atexit:
+            atexit.register(_close_session_atexit)
```

Comment:
The implementation of `_close_session_atexit` looks like it could handle when `_active_sessions` is empty. Can we do this registration at a module level?

Reply:
Yes, it can be simpler.